main.cpp

```diff
@@ -0,0 +1,15 @@
+#include <iostream>
+
+#include <gamemanager.h>
+
+int main()
+{
+  GameManager* game_manager = game_manager->Instance();
+
+  std::shared_ptr<Character> Maple(new Character("Maple", 150, 50));
+  std::shared_ptr<Character> Sally(new Character("Sally", 45, 30));
```

Comment:
Az is erdekelne meg mindig, hogy miert hasznals shared ptr-t. Nem baj, csak indokold meg, hogy az alternativanal Neked miert jobb ez. En nem latok ra okot, de lehet Te latsz.

Reply:
azért akartam okos mutatót használni, mert akkor nem kell nekem felszabadítani a memóriát, de közben rájöttem hogy létre lehet hozni a Character osztályt úgy is, hogy nem mutatóként. Ezért kivettem ezeket.